__tests__/app.test.js

```diff
@@ -379,3 +379,39 @@ describe("GET /api/users", () => {
       });
   });
 });
+describe("GET /api/articles", () => {
+  test("should be available on /api/articles?topic=mitch", () => {
+    return request(app).get("/api/articles?topic=mitch").expect(200);
+  });
+  test("should respond with an array of article objects, each of which should have the expected properties", () => {
+    const validTopic = "mitch";
+    return request(app)
+      .get(`/api/articles?topic=${validTopic}`)
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toHaveLength(4);
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: validTopic,
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  test("status:404, responds with an error message when the topic does not exist", () => {
+    const nonExistingTopic = 'karo';
+    return request(app)
+      .get(`/api/articles?topic=${nonExistingTopic}`)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found!");
+      });
```

Comment:
Could we have one more test here please? What would happen if it was a valid topic but had no articles?